doc/examples/perception_pipeline/perception_pipeline_tutorial.rst

```diff
@@ -1,198 +1,99 @@
 
-Detecting and Adding Object as Collision Object
------------------------------------------------
+For the commands in Shell 2, we must save both camera topics and tf topics due to the fact that MoveIt perception pipeline needs to listen TF's in order to convert the coordinates of pointcloud points according to ``world`` frame. Moreover, the reason of publishing static tf from world to camera frames in ``depth_camera_environment.launch.py`` is that it's necessary to determine transformation between robot and poincloud and that MoveIt's sensor plugins uses this transformation later.
 
-In this section, we will demonstrate an example of extracting a cylinder from a pointcloud, computing relevant values and adding it as a collision object to the planning scene.
-We will be working with point clouds but it can be implemented similarly with depth maps.
+By the way, you can also use :codedir:`this rviz file <examples/perception_pipeline/rviz2/depth_camera_environment.rviz>` on Github to visualize poincloud topics in rviz.
 
-After running the code, you should be able to see something like this in rviz:
+In next step, it will be used this recorded bag file for octomap creation.
 
-.. image:: cylinder_collision_object.png
-   :width: 700px
 
-Running the Code
-++++++++++++++++
-Roslaunch the launch file to run the code directly from moveit_tutorials: ::
+Configuration For 3D Sensors
+----------------------------
+Now whatever you get bag file, you will see ``/camera_1/points``, ``/camera_2/points``, ``/tf`` and ``/tf_static`` in both way when played bag file. It should be created following config file for MoveIt to process these pointcloud topics in planning pipeline. You can also go to :codedir:`here <examples/perception_pipeline/config/sensors_3d.yaml>` for seeing all ``sensors_3d.yaml`` config file on Github.
 
- roslaunch moveit_tutorials detect_and_add_cylinder_collision_object_demo.launch
+.. tutorial-formatter:: config/sensors_3d.yaml
 
-KNOWN ISSUE - You may see the following error when running the demo ::
+sensors_3d.yaml: ::
 
-  ros.moveit_ros_planning.planning_scene_monitor: Transform error: Lookup would require extrapolation into the future.  Requested time 1527473962.793050157 but the latest data is at time 1527473962.776993978, when looking up transform from frame [panda_link2] to frame [camera_rgb_optical_frame]
-  ros.moveit_ros_perception: Transform cache was not updated. Self-filtering may fail.
+    sensors:
+      - camera_1_pointcloud
+      - camera_2_pointcloud
+    camera_1_pointcloud:
+        sensor_plugin: occupancy_map_monitor/PointCloudOctomapUpdater
+        point_cloud_topic: /camera_1/points
+        max_range: 5.0
+        point_subsample: 1
+        padding_offset: 0.1
+        padding_scale: 1.0
+        max_update_rate: 1.0
+        filtered_cloud_topic: /camera_1/filtered_points
+    camera_2_pointcloud:
+        sensor_plugin: occupancy_map_monitor/PointCloudOctomapUpdater
```

Comment:
How much work would it be to add a depthcamera example here instead of the second Pointcloud updater (and can you briefly explain why we need two if that's required)?

Reply:
It's not necessary to use two pointcloud, These pointclouds show both two sides of table. We can turn second plugin into depth camera plugin. When i completed, i will tag you.

Reply:
I've added. You can find some configurations here. 

https://github.com/CihatAltiparmak/moveit2_tutorials/blob/c5383b570f8e03157165f878dc67f27c6ec3df28/doc/examples/perception_pipeline/config/sensors_3d.yaml#L1-L22